app/models/service_template.rb

```diff
@@ -545,13 +545,7 @@ def generic_custom_buttons
   end
 
   def adjust_service_type
-    svc_type = self.class::SERVICE_TYPE_ATOMIC
-    service_resources.try(:each) do |sr|
-      if sr.resource_type == 'Service' || sr.resource_type == 'ServiceTemplate'
-        svc_type = self.class::SERVICE_TYPE_COMPOSITE
-        break
-      end
-    end
+    svc_type = service_resources.any? { |st| st.resource_type.in?(['Service', 'ServiceTemplate']) } ? self.class::SERVICE_TYPE_COMPOSITE : self.class::SERVICE_TYPE_ATOMIC
```

Comment:
> The `try(:each)` above seems strange ... is it possible `service_resources` doesn't respond to `each`?

It's a `has_many` relation, so I don't see how it could be `nil`.

> Also, couldn't this line set the `self.service_type` directly?

Done